megablocks/layers/moe.py

```diff
@@ -200,6 +204,11 @@ def permute_and_compute(
             x, indices, expert_weights, bins, top_k)
 
     def forward_once(self, x, expert_weights, top_experts):
+        # Expert dropout: Randomly drop experts based on the dropout rate
+        if self.training and self.args.expert_dropout_rate > 0:
+            dropout_mask = torch.rand_like(expert_weights) < self.args.moe_expert_dropout_rate
+            top_experts = top_experts.masked_fill(dropout_mask, -1)
```

Comment:
I'm not certain that assigned these tokens to expert '-1' will have the desired effect here...

Also, I'd understood that the "expert dropout" method from Switch Transformer was just turning up the dropout rate in the FFN, rather than dropping whole experts? Is this something different?

Reply:
I just looked at the paper and you are right. I overfit to the name, sorry!